packages/devtools_app/lib/src/debugger/flutter/console.dart

```diff
@@ -3,24 +3,85 @@
 // found in the LICENSE file.
 
 import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
 
 import '../../flutter/common_widgets.dart';
+import '../../flutter/theme.dart';
 import 'common.dart';
+import 'debugger_controller.dart';
+
+// TODO(devoncarew): Allow scrolling horizontally as well.
+
+// TODO(devoncarew): Show some small UI indicator when we receive stdout/stderr.
+
+// TODO(devoncarew): Support hyperlinking to stack traces.
+
+/// Display the stdout and stderr output from the process under debug.
+class Console extends StatefulWidget {
+  const Console({
+    Key key,
+    this.controller,
+  }) : super(key: key);
+
+  final DebuggerController controller;
+
+  @override
+  _ConsoleState createState() => _ConsoleState();
+}
+
+class _ConsoleState extends State<Console> {
+  final scrollController = ScrollController();
 
-class Console extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
+    final textStyle =
+        theme.textTheme.bodyText2.copyWith(fontFamily: 'RobotoMono');
 
     return OutlinedBorder(
       child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
         children: [
           debuggerSectionTitle(theme, text: 'Console'),
-          const Expanded(
-            child: Center(child: Text('todo:')),
+          Expanded(
+            child: Padding(
+              padding: const EdgeInsets.all(denseSpacing),
+              child: SingleChildScrollView(
+                controller: scrollController,
+                child: ValueListenableBuilder(
+                  valueListenable: widget.controller.stdio,
+                  builder: (context, value, _) {
+                    // If we're at the end already, scroll to expose the new
+                    // content.
+                    final pos = scrollController.position;
+                    if (pos.pixels == pos.maxScrollExtent) {
+                      SchedulerBinding.instance.addPostFrameCallback((_) {
+                        _scrollToBottom();
+                      });
+                    }
+
+                    return Text(value, style: textStyle);
+                  },
+                ),
+              ),
+            ),
           ),
         ],
       ),
     );
   }
+
+  void _scrollToBottom() async {
+    await scrollController.animateTo(
+      scrollController.position.maxScrollExtent,
+      duration: const Duration(milliseconds: 100),
+      curve: Curves.linear,
+    );
+
+    // Scroll again if we've received new content in the interim.
+    final pos = scrollController.position;
```

Comment:
wouldn't this case would also trigger if the user interacted with the scrollable view in which case I don't think we would want to jump to the bottom?
It could be worth it to create a common widget that handles the auto-scroll-to-bottom case.

Reply:
> wouldn't this case would also trigger if the user interacted with the scrollable view in which case I don't think we would want to jump to the bottom?

Yes, but it would be tough to distinguish between when we added more content and when the user manually scrolled.

What this is guarding against is us stopping auto-scrolling because we ourselves added more content before this animation completes (which I did see when testing).

I can add a todo: here to improve the behavior when the user manually scrolls, or think about ways to address the issue.

> It could be worth it to create a common widget that handles the auto-scroll-to-bottom case.

It would be nice to have for the logging view as well. Perhaps we can look at generalizing when we do look to implement it in one more place?

Reply:
Yes would be great to have a generalized robust scroll to bottom view.
Would also be nice to have a light weight abstraction on top of ScrollController that supports some of the cases we need here as well as in the inspector. The default implementation isn't good when you need to support scrolling to a new location when an animation is in progress and when you need to know the end destination of the in progress scroll.

Reply:
I added a TODO: here to generalize the auto-scroll-to-bottom work.